indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -76,10 +78,21 @@
 
   @Inject
   public TaskLockbox(
-      TaskStorage taskStorage
+      TaskStorage taskStorage,
+      ServerConfig serverConfig
+  )
+  {
+    this.taskStorage = taskStorage;
+    this.lockTimeoutMillis = serverConfig.getMaxIdleTime().getMillis();
+  }
+
+  public TaskLockbox(
+      TaskStorage taskStorage,
+      long lockTimeoutMillis
```

Comment:
> For now serverConfig.getMaxIdleTime() is the default timeout b/c even if task gets a lock after this period, overlord can not write the response in the closed socket.

Using maxIdleTime as a default value sounds good. 

> Yes for now, it can also be extended where client can send a timeout.

I think we need to set different timeouts for each lock request in very near future because I'm working on prioritized locking (https://github.com/druid-io/druid/issues/4479, https://github.com/druid-io/druid/pull/1679) and this timeout feature will be great if tasks can set different timeouts according to their priorities.